### PR TITLE
Remove deprecations scheduled for removal in pytest 7.1

### LINKED ISSUE #8838: Remove deprecated 7.0 features
Remove the features that are deprecated and were changed into errors in `7.0`.

Follow up to #7363.

Note: skip `test_deprecation_warning_as_error` again (#8833) so it can be reused in `8.0`.

### PR DESCRIPTION
**Pull request by bluetech**
Fixes #8838. Split into commits for each removal.

### CODE REVIEW

**Review by nicoddemus — APPROVED** [review]
Great work!

File: changelog/8838.removal.rst [hunk 1]
```diff
@@ -0,0 +1,17 @@
+As per our policy, the following features have been deprecated in the 6.X series and are now
```
Comment (nicoddemus) [inline changelog/8838.removal.rst:1]: This file should be named `8838.breaking.rst`.

Comment (bluetech) [reply changelog/8838.removal.rst]: Right, fixed.

File: changelog/8838.breaking.rst [hunk 2]
```diff
@@ -0,0 +1,17 @@
+As per our policy, the following features have been deprecated in the 6.X series and are now
+removed:
+
+* ``pytest._fillfuncargs`` function.
+
+* ``pytest_warning_captured`` hook - use ``pytest_warning_recorded`` instead.
+
+* ``message`` parameter of ``pytest.raises``.
```
Comment (nicoddemus) [inline changelog/8838.breaking.rst:8]: Just realized I didn't see this removal, did I miss something?

Comment (bluetech) [reply changelog/8838.breaking.rst:8]: Good catch, I copy/pasted from the 6.x PR, left this by mistake.

**Pull request merged**